src/argilla/client/feedback/metrics/annotator_metrics.py

```diff
@@ -0,0 +1,251 @@
+                metrics[user_id].append(AnnotatorMetricResult(metric_name=metric_name, result=result))
+
+        return metrics
+
+
+class AnnotatorMetricResult(BaseModel):
+    metric_name: str
+    result: Union[float, Dict[str, float], pd.DataFrame]
+
+    class Config:
+        arbitrary_types_allowed = True
+
+
+class AnnotatorMetricBase(ABC):
+    def __init__(self, responses=None, suggestions=None) -> None:
+        self._responses = responses
+        self._suggestions = suggestions
+
+    def compute(self, **kwargs):
+        responses, suggestions = self._pre_process(self._responses, self._suggestions)
+        result = self._compute(responses, suggestions, **kwargs)
+        return self._post_process(result)
+
+    def _pre_process(self, responses, suggestions) -> Any:
+        return responses, suggestions
+
+    def _post_process(self, result: Any) -> Any:
+        return result
+
+    @abstractmethod
+    def _compute(self, responses, suggestions, **kwargs):
+        pass
+
+
+def is_multiclass(data) -> bool:
+    return len(np.unique(data)) > 2
+
+
+class AccuracyMetric(AnnotatorMetricBase):
+    def _compute(self, responses, suggestions, **kwargs):
+        return accuracy_score(responses, suggestions, **kwargs)
+
+
+class F1ScoreMetric(AnnotatorMetricBase):
+    def _compute(self, responses, suggestions, **kwargs):
+        if is_multiclass(responses) or is_multiclass(suggestions):
+            if not kwargs.get("average"):
+                kwargs.update({"average": "macro"})
+        return f1_score(responses, suggestions, **kwargs)
+
+
+class PrecisionMetric(AnnotatorMetricBase):
+    def _compute(self, responses, suggestions, **kwargs):
+        if is_multiclass(responses) or is_multiclass(suggestions):
+            if not kwargs.get("average"):
+                kwargs.update({"average": "macro"})
+        return precision_score(responses, suggestions, **kwargs)
+
+
+class RecallMetric(AnnotatorMetricBase):
+    def _compute(self, responses, suggestions, **kwargs):
+        if is_multiclass(responses) or is_multiclass(suggestions):
+            if not kwargs.get("average"):
+                kwargs.update({"average": "macro"})
+        return recall_score(responses, suggestions, **kwargs)
+
+
+class ConfusionMatrixMetric(AnnotatorMetricBase):
+    def _compute(self, responses, suggestions, **kwargs):
+        if is_multiclass(responses) or is_multiclass(suggestions):
+            return multilabel_confusion_matrix(responses, suggestions, **kwargs)
+        return confusion_matrix(responses, suggestions, **kwargs)
+
+
+class CorrelationCoefficientMetric(AnnotatorMetricBase):
+    def _compute(self, responses, suggestions, kind: Literal["pearson", "spearman", "kendall"] = "spearman", **kwargs):
+        return (
+            pd.DataFrame({"responses": responses, "suggestions": suggestions})
+            .corr(method=kind)["responses"]
+            .loc["suggestions"]
+        )
+
+
+@requires_dependencies("evaluate")
```

Comment:
@dvsrepo what do you think? Does it work to have things like `require_dependencies` and not force them as a required install for the package? Or would you rather have us reproduce the logic for these metrics?